.github/workflows/integration.yaml

```diff
@@ -0,0 +1,107 @@
+name: 'Integration'
+
+on:
+  push:
+    branches:
+    - 'main'
+  pull_request:
+    branches:
+    - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+permissions:
+  contents: 'read'
+  id-token: 'write'
+
+jobs:
+  test:
+    if: ${{ github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name && github.actor != 'dependabot[bot]' }}
+    runs-on: 'ubuntu-latest'
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - name: 'image'
+          image: 'gcr.io/cloudrun/hello:latest'
+
+    name: 'from_${{ matrix.name }}'
+
+    steps:
+    - id: 'checkout'
+      name: 'Checkout'
+      uses: 'actions/checkout@v3'
+
+    - id: 'compute-test-uid'
+      name: 'Compute test UID'
+      run: |-
+        echo "TEST_UID=${GITHUB_JOB}-${{ matrix.name }}-${GITHUB_SHA::7}-${GITHUB_RUN_NUMBER}" >> ${GITHUB_ENV}
+
+    - id: 'set-up-node'
+      name: 'Set up Node'
+      uses: 'actions/setup-node@v3'
+      with:
+        node-version: '16.x'
+
+    - id: 'build'
+      name: 'Build'
+      run: 'npm ci && npm run build'
+
+    - id: 'auth'
+      name: 'Auth'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: '${{ secrets.WIF_PROVIDER_NAME }}'
+        service_account: '${{ secrets.SERVICE_ACCOUNT_EMAIL }}'
+
+    - id: 'gcloud-setup'
+      name: 'Set up Cloud SDK'
+      uses: google-github-actions/setup-gcloud@v0
+      with:
+        project_id: ${{ secrets.PROJECT_ID }}
+
+    - id: 'configure-pipeline'
+      name: 'Configure deployment pipeline'
+      run: 'for template in $(ls tests/fixtures/*.template); do envsubst < ${template} > ${template%.*} ; done'
```

Comment:
It'd probably be better to do this as a multi-line run:

```yaml
run: |-
  for template in ...
```

It would also be helpful to understand exactly what this is doing, perhaps adding a comment for future developers to understand.

Reply:
+1, seems like the for loop might not be needed currently for a single template?

Reply:
I simplified the logic and added a clearer description now that only a single file is templated.